index.rst

```diff
@@ -265,6 +306,39 @@ If the first point (direct use of JWTs by third-party services) becomes compelli
 The primary driver for using opaque tokens rather than JWTs is length, which in turn is driven by the requirement to support HTTP Basic authentication.
 If all uses of HTTP Basic authentication can be shifted to token authentication and that requirement dropped, the decision to use opaque tokens rather than JWTs should be revisited (but revocation would need to be addressed).
 
+Token scope
+^^^^^^^^^^^
+
+JWTs natively carry their scope as one of their claims.
+For opaque bearer tokens, the scope will be stored as part of the corresponding session entry.
+In either case, tokens can be restricted to specific scopes.
+
+The token used for web browser authentication and stored in the session cookie should have unlimited scope.
+The strongest authentication of the user is via the web browser, and the user should be able to take whatever actions they are allowed to do via that path.
+
+More limited scopes are therefore for user-created API tokens, and possibly for auto-created API tokens for automatic token delegation (such as for the notebook environment).
+The details of automated token delegation are intentionally left open in this design, apart from ensuring it will be possible to support them, since the requirements and use cases are not yet clear.
+
+For user-created API tokens, there will be a balance between the security benefit of more restricted-use tokens and the UI complexity of giving the user a lot of options when creating a token.
+The current design intention is to identify the use cases where a user would need a separate API token and provide a UI to create a token for that specific purpose, possibly with an advanced user escape hatch that would allow the user to make a more unrestricted choice of token scopes.
+This balance will be discussed in more detail in future documents as the requirements become clearer.
+
+Federated authentication
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The simplest design for API authentication, and indeed for the entire authentication system, is for all protected applications to run within the same Kubernetes cluster.
+Each separate Kubernetes cluster (the summit, for example) would have its own independent instantiation of this authentication system, without any cross-realm authentication.
+
+This may not be practical or desirable in practice.
```

Comment:
It's worth saying in this context that different R/LSP instances (clusters?) may have different sets of authorized users.  Cross-cluster working capabilities would have to take this into account.